app/src/org/commcare/tasks/ConnectionDiagnosticTask.java

```diff
@@ -70,8 +64,9 @@ public ConnectionDiagnosticTask(Context c) {
     @Override
     protected Test doTaskBackground(Void... params) {
         Test out = null;
-        if (!isOnline(this.c)) {
+        if (!ConnectivityStatus.isNetworkAvailable(this.c)) {
             out = Test.isOnline;
+            Logger.log(CONNECTION_DIAGNOSTIC_REPORT, noNetwork);

```

Comment:
we need to log both positive and negative scenarios here and should utilise already existing strings to do it `logNotConnectedMessage` and `logConnectionSuccessMessage` instead of adding new ones.